### ISSUE TITLE
[BUG] Unable to Assign FAN_PIN to E0_AUTO_FAN_E0_PIN for Auto Fan Feature with Single Fan Printers

### ISSUE THREAD

**Issue by borland1** [MarlinFirmware/Marlin#18813]
### Bug Description
Enabling extruder Auto Fan feature in Configuration_adv.h as follows..

```
#define E0_AUTO_FAN_PIN 9  // Fan Connector (pin D9) on MKS_GEN_L board
```

produces` SanityCheck.h` error: `"You cannot set E0_AUTO_FAN_PIN equal to FAN_PIN."` and so compiler fails to compile FW bin file.

This is for single extruder, single fan, printer.

I have tried disabling FAN_PIN by overriding pin definition file `pins_RAMPS.h`,  and enabling Auto Fan, both in Configuration_adv.h as follows...
```
#define FAN_PIN -1
#define E0_AUTO_FAN_PIN 9
```
and this compiles with success, and Auto Fan works, but Fan Speed cannot be set with LCD menu.  LCD menu also shows two Fans when only one is configured ( see issue #18529 ).

### My Configurations

Config files ( bug-fix-2.0.x ):

[ConfigFiles.zip](https://github.com/MarlinFirmware/Marlin/files/4984857/ConfigFiles.zip)

**Comment by borland1** [comment]
I found a workaround for this bug where the LCD's Temperature menu showing two Fans, when only one is configured.  I posted the workaround in issue #18529 .  With both workarounds (above mentioned fix, plus auto fan feature fix ), no fan is shown on LCD's Temperature menu.  
I guess that is the intended function of Auto Fan ( i.e., Not able to be manually set ).  Would be nice if the Auto Fan feature was more dynamic, in allowing LCD menu manual adjusting above the EXTRUDER_AUTO_FAN_SPEED value and by having a more dynamic EXTRUDER_AUTO_FAN_SPEED value that is dependent on the hotend temperature setting.
